File: solutions/palindrome-permutation/palindrome-permutation.c
// @Title: Palindrome Permutation
// @Author: colinjxc
// @Date: 2022-02-01T03:35:08+08:00
// @URL: https://leetcode-cn.com/problems/palindrome-permutation


bool canPermutePalindrome(char * s){
    int seen[128] = {0,};
    while (*s) seen[*s++]++;

    int cnt = 0;
    for (int i = 0; i < sizeof(seen)/sizeof(seen[0]); i++) {
        cnt += seen[i] % 2;
        if (cnt > 1) return false;
    } 
    return true;
}

File: solutions/palindrome-permutation/palindrome-permutation.py
// @Title: Palindrome Permutation
// @Author: colinjxc
// @Date: 2022-02-01T03:26:00+08:00
// @URL: https://leetcode-cn.com/problems/palindrome-permutation


class Solution:
    def canPermutePalindrome(self, s: str) -> bool:
        seen = collections.defaultdict(int)
        for c in s:
            seen[c] += 1
        cnt = 0
        for v in seen.values():
            cnt += v % 2
            if cnt > 1:
                return False
        return True
